### PR TITLE
feat: add empty result, search result and search placeholder

### PR DESCRIPTION
**Pull request by joonchoi0918**

### CODE REVIEW

File: src/components/AsyncTable/AsyncTable.module.scss [hunk 1]
```diff
@@ -38,4 +38,10 @@
...
   list-style: none;
   grid-gap: 0.875rem 0.75rem;
-}
\ No newline at end of file
+}
+
+.Empty {
+  margin-top: 2rem;
+  text-align: center;
+  font-style: italic;
```
Comment (domw30) [inline src/components/AsyncTable/AsyncTable.module.scss:46]: we could probs add font-style to variables. Unless we're waiting for theme engine stuff

File: src/components/AsyncTable/AsyncTable.tsx [hunk 2]
```diff
@@ -114,7 +117,12 @@ export const AsyncTable = <T extends unknown>({
 
   // Grid view doesn't render Skeletons when loading (yet)
   if (isGridView && isLoading) {
-    return <LoadingIndicator className={styles.Loading} text={'Loading'} />;
+    return <LoadingIndicator className={styles.Loading} text={loadingText ?? 'Loading'} />;
+  }
+
+  // Show emotyText when there is no data
```
Comment (domw30) [inline src/components/AsyncTable/AsyncTable.tsx:123]: nit: typo `emotyText` -> `emptyText`

**Review by domw30 — APPROVED** [review]
Nice Joon 🙌

File: src/components/AsyncTable/AsyncTable.tsx [hunk 3]
```diff
@@ -124,18 +132,25 @@ export const AsyncTable = <T extends unknown>({
...
         />
       )}
-      <InfiniteScrollWrapper className={className}>
-        {isGridView ? (
-          <Grid cards={chunkedComponents} isSingleColumnGrid={isSingleColumnGrid} />
-        ) : (
-          <Table rows={chunkedComponents} columns={columns} isLoading={isLoading} />
-        )}
-      </InfiniteScrollWrapper>
+      {query && !isLoading && filteredData.length ? (
```
Comment (eric-zero-dev) [inline src/components/AsyncTable/AsyncTable.tsx:141]: can we use another variable here?

```
const hasItems = query && !isLoading && filteredData.length;
```

Hmmm, by the way, if `query` is empty we show empty message? -> `query && ... ?  <table> : <empty>`?

**Review by eric-zero-dev — CHANGES_REQUESTED** [review]

File: src/components/AsyncTable/AsyncTable.module.scss (hunk 1 above)
Comment (joonchoi0918) [reply src/components/AsyncTable/AsyncTable.module.scss:46]: Copied from zDaos table, but I think we don't need `italic` font style

**Review by eric-zero-dev — APPROVED** [review]
LGTM!

**Pull request merged**